helix-core/src/main/java/org/apache/helix/monitoring/mbeans/JobMonitor.java

```diff
@@ -41,6 +45,12 @@
   private static final Logger LOG = LoggerFactory.getLogger(JobMonitor.class);
   private static final long DEFAULT_RESET_INTERVAL_MS = 60 * 60 * 1000; // 1 hour
 
+  // The root path of ZNodes that are used to sync certain metrics to ZooKeeper. This path is the
+  // first level, and its children nodes will represent each job type. Since the metrics are
+  // aggregated by job types, the metric values will be recorded in each ZNode belonging to a job
+  // type as simple fields.
+  private static final String SYNC_ZK_ROOT_PATH = "/JOB_MONITOR_METRICS";
```

Comment:
Why we need a ZNode to persist metrics data?

Reply:
Is there a reason we don't use MBean for such metrics?

Reply:
Right now the metrics are emitted by the controller. We want to expose the metrics to the client side, therefore ZNodes are used as a medium to store the metrics for access. 

MBean is still used; the ZNodes are for "syncing" or "copying" the metrics to client side.

Reply:
It's now necessary for doing in this way:
1. These metrics can be done in controller by looking at the stats of the ZNode property.
2. Even if you would like to have it in client side. You should make it reported in MBean since all the data can be derived from config/context. So ingraph can fetch MBean to report it. This is not a good idea to have metrics reporting to introduce new ZNode create/read/write.

Reply:
I'm not trying to say no to this idea before I fully understand the context here. But,
1. Who are this "We"?
2. Do we have a design that illustrates the motivation and potential risks? I would like to take a look and see if there are any other ways.

Reply:
I would not think it is a good idea to use ZK as syncing store for metrics.. ZK is not designed for this purpose. ZK is such a core part of infra. Syncing metrics to ZK definitely increase read/write traffic to ZK, which may impact the other critical services. If we have a better design not to use ZK as a syncing store, exposing metrics to client is fine.

Could you explain any case "users that cannot rely directly on the controller emitted metrics to see the metric values."?  I would like to see if there is an alternative solution that doesn't use ZK.